tests/mode_test.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/python3
+
+# Test that we can read the sensor modes, and then configure the camera
+# into each mode and get the correct framerate
+
+from picamera2 import Picamera2
+import time
+from math import isclose
+
+picam2 = Picamera2()
+
+for mode in picam2.sensor_modes:
+    picam2.video_configuration = picam2.create_video_configuration(
+        raw={"size": mode["size"], "format": mode["format"].format},
+    )
+    picam2.configure("video")
+    picam2.set_controls({"FrameRate": int(mode["fps"])})
+    picam2.start()
```

Comment:
After calling `configure` I think we could retrieve the current configuration (`picam2.camera_configuration()`) and check what's in the raw stream. For each mode we could configure the camera using both the "normal" (unpacked) and packed formats.